duckduckgo/iOS
# Add Maestro

Task/Issue URL: https://app.asana.com/0/42792087274227/1203635285285764/f
Tech Design URL:
CC:

**Description**:
This PR introduces nightly e2e tests using Maestro. The first set of tests are a porting of the e2e set used on Android.



**Steps to test this PR**:
1. Make sure CI is green

To test locally:
1. Install Maestro follow [their instructions](https://maestro.mobile.dev/getting-started/installing-maestro). (make sure to install IDB tool as well)
1. Install Fastlane and run `setup_e2e_tests` lane: `bundle install && bundle exec fastlane setup_e2e_tests`. This will copy the shared modules to subfolders of each test suite and it's required because Maestro YAML parser doesn't seem to understand relative paths. 
2. Build a testable version of the app and install on your simulator. 
3. With the simulator running, use `maestro test` to run the tests. You can run single suites with `maestro test .maestro/<suite folder>` or single tests with `maestro test .maestro/<suite folder>/<test name>` 



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
